### PR TITLE
Bump CI jobs to Python 3.12

### PR DESCRIPTION
**Pull request by jacobtylerwalls**

### CODE REVIEW

**Comment by Pierre-Sassoulas** [comment]
Blocked by 3.12 being released in the github available runners (can't find the repo but I'll link it later because I'm watching it and I'll get notified when 3.12 is up)

**Comment by Pierre-Sassoulas** [comment]
It's https://github.com/actions/python-versions 

https://github.com/pylint-dev/pylint/pull/9093#issuecomment-1743420633

File: .readthedocs.yaml [hunk 1]
```diff
@@ -8,7 +8,7 @@ sphinx:
 build:
   os: ubuntu-22.04
   tools:
-    python: "3.11"
+    python: "3.12"
```
Comment (Pierre-Sassoulas) [inline .readthedocs.yaml:11]:
```suggestion
    python: "3.11"
```

**Comment by Pierre-Sassoulas** [comment]
Same pypi 3.10 bug is happening elsewhere : https://github.com/pylint-dev/astroid/pull/2316 I think main is broken.

**Comment by jacobtylerwalls** [comment]
OK to merge this then? Don't want to spend any open source banana sundae points with a self-merge 😉

**Comment by jacobtylerwalls** [comment]
I'll delete the cache and rerun it.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Comment by Pierre-Sassoulas** [comment]
> Don't want to spend any open source banana sundae points with a no-approval-merge 

In truth the sweet banana sundae point (and money for increasing said points) for well reviewed code is no more :(. But sometime I think a review is not necessary and then I break main in multiple unexpected ways or a review happen and something obvious is pointed out so review is always better (but we don't have to be zealots about it either)

**Pull request merged**